src/main/java/seedu/address/model/person/WardNumber.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the Ward Number of an patient in the database.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidWardNumber(Integer)}
+ */
+public class WardNumber {
+    public static final String MESSAGE_CONSTRAINTS = "Ward Number is a positive integer for inpatient patients "
+            + "and blank for outpatient patients";
+
+    public final Integer value;
+
+    /**
+     * Constructs an {@code Ward Number}.
+     *
+     * @param wardNumber A valid ward number.
+     */
+    public WardNumber(Integer wardNumber) {
+        checkArgument(isValidWardNumber(wardNumber), MESSAGE_CONSTRAINTS);
+        value = wardNumber;
+    }
```

Comment:
Using `int` over the wrapper type `Integer` might help with object housekeeping, but that can be ignored for now.

Reply:
Agree, was considering using int, will change it in future or just use string with regex